### DISCUSSION TITLE
uv_udp_send send tow packets each time (not one by one)

### DISCUSSION THREAD

**Discussion by 5415nofar1** [libuv/libuv#3274] [Q&A]
When we sent a packet with uv_udp_send this is not really send until the second packet will came and then two packets together sent to the network.

When you open Wireshark you can see it easily - after two packages arrive they are sent together to the network

This cause latency issues.

**Comment by bnoordhuis** [comment]
I'm not sure what you mean. Can you post some screenshots or something that shows what you're seeing?

**Reply by ydanzige** [reply]
I add the code that I wrote and demonstrate the problem, and the pcap file that you can see that each time there is two packets received and then two packet will send

[main.txt](https://github.com/libuv/libuv/files/7059464/main.txt)
![pcap_libuv](https://user-images.githubusercontent.com/32995650/130986485-546b2ddf-6735-4043-9848-8c9f02dbf168.png)

**Reply by santigimeno** [reply]
I'm not able to reproduce what you're seeing. Can you exactly explain what you're doing?

**Reply by ydanzige** [reply]
In my small program I receive messages from network on port 18080 and send them to port 18083. As you can see in the wireshark, instead of receiving one message and sending it ,the libuv waits until he has two messages and then send both of them, thus introducing latency for the first one.

**Comment by ydanzige** [comment]
Hi, could you please have a look at the issue? It is very urgent for me.
Thanks a lot!
